week3/binary/maverickchaser/Main.hs

```diff
@@ -0,0 +1,12 @@
+bin :: Int -> String
+bin 0 = ""
+bin x = bin (x `div` 2) ++ show (x `mod` 2)
```

Comment:
this is going to be O(n²) btw (not bad but could be better if you used an accumulator)